fe/src/main/java/org/apache/doris/clone/DynamicPartitionScheduler.java

```diff
@@ -0,0 +1,200 @@
+import org.apache.doris.catalog.HashDistributionInfo;
+import org.apache.doris.catalog.OlapTable;
+import org.apache.doris.catalog.RangePartitionInfo;
+import org.apache.doris.catalog.Table;
+import org.apache.doris.catalog.TableProperty;
+import org.apache.doris.common.DdlException;
+import org.apache.doris.common.Pair;
+import org.apache.doris.common.util.DynamicPartitionUtil;
+import org.apache.doris.common.util.MasterDaemon;
+import org.apache.doris.common.util.TimeUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class DynamicPartitionScheduler extends MasterDaemon {
+    private static final Logger LOG = LogManager.getLogger(DynamicPartitionScheduler.class);
+
+    private static final String defaultValue = "N/A";
+    private static String lastSchedulerTime = defaultValue;
+    private static String lastUpdateTime = defaultValue;
+    private static State dynamicPartitionState = State.NORMAL;
+    private static String Msg = defaultValue;
+
+    public enum State {
+        NORMAL,
+        ERROR
+    }
+
+    public static String getLastSchedulerTime() {
+        return lastSchedulerTime;
+    }
+
+    public static void setLastSchedulerTime(String lastSchedulerTime) {
+        DynamicPartitionScheduler.lastSchedulerTime = lastSchedulerTime;
+    }
+
+    public static String getLastUpdateTime() {
+        return lastUpdateTime;
+    }
+
+    public static void setLastUpdateTime(String lastUpdateTime) {
+        DynamicPartitionScheduler.lastUpdateTime = lastUpdateTime;
+    }
+
+    public static State getDynamicPartitionState() {
+        return dynamicPartitionState;
+    }
+
+    public static void setDynamicPartitionState(State state) {
+        dynamicPartitionState = state;
+    }
+
+    public static String getMsg() {
+        return Msg;
+    }
+
+    public static void setMsg(String msg) {
+        Msg = msg;
+    }
+
+
+    private static Set<Pair<Long, Long>> dynamicPartitionTableInfo = new HashSet<>();
+    private boolean initialize;
+
+    public DynamicPartitionScheduler(String name, long intervalMs) {
+        super(name, intervalMs);
+        this.initialize = false;
+    }
+    public synchronized static void registerDynamicPartitionTable(Long dbId, Long tableId) {
+        dynamicPartitionTableInfo.add(new Pair<>(dbId, tableId));
+    }
+
+    public synchronized static void removeDynamicPartitionTable(Long dbId, Long tableId) {
+        dynamicPartitionTableInfo.remove(new Pair<>(dbId, tableId));
+    }
+
+
+
+    private void dynamicAddPartition() {
+        for (Pair<Long, Long> tableInfo : dynamicPartitionTableInfo) {
+            Long dbId = tableInfo.first;
+            Long tableId = tableInfo.second;
+            Database db = Catalog.getInstance().getDb(dbId);
```

Comment:
hold database's lock before visiting members inside the database.
This is mentioned in my last review.